File: backend/app/schemas/calendar.py

from typing import Optional, List
from pydantic import BaseModel
import datetime
import uuid

class EventBase(BaseModel):
    title: str
    description: Optional[str] = None
    start_time: datetime.datetime
    end_time: datetime.datetime

class EventCreate(EventBase):
    collaborator_ids: Optional[List[uuid.UUID]] = []

class EventUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    start_time: Optional[datetime.datetime] = None
    end_time: Optional[datetime.datetime] = None
    collaborator_ids: Optional[List[uuid.UUID]] = []

class Event(EventBase):
    id: uuid.UUID
    owner_id: uuid.UUID
    collaborators: List[dict] = []  # Simplified to avoid circular imports

    class Config:
        from_attributes = True

File: client/src/types/calendar.ts
export interface CalendarEvent {
  id: string;
  title: string;
  description?: string;
  start_time: string;
  end_time: string;
  owner_id: string;
  collaborators?: {
    id: string;
    username: string;
  }[];
}

export interface EventCreate {
  title: string;
  description?: string;
  start_time: string;
  end_time: string;
  collaborator_ids?: string[];
}

export interface ConflictSlot {
  start: string;
  end: string;
}

export interface ConflictDetail {
  message: string;
  available_slots: ConflictSlot[];
}
